src/main/java/com/microsoft/cse/helium/app/controllers/ActorsController.java

```diff
@@ -45,14 +47,13 @@
       return actorsDao
           .getActorById(actorId)
           .map(savedActor -> ResponseEntity.ok(savedActor))
-          .defaultIfEmpty(ResponseEntity.notFound().build());
+          .switchIfEmpty(Mono.defer(() -> 
+            Mono.error(new ResponseStatusException(HttpStatus.NOT_FOUND, "Actor not found"))));
     } else {
       logger.error("Invalid Actor ID parameter " + actorId);
-      return Mono.justOrEmpty(
-          ResponseEntity.badRequest()
-              .contentType(MediaType.TEXT_PLAIN)
-              .header("Invalid Actor ID parameter")
-              .build());
+
+      return Mono.error(new ResponseStatusException(
+        HttpStatus.BAD_REQUEST, "Invalid Actor ID parameter"));
```

Comment:
How does the MediaType look for this? Is it TEXT_PLAIN?

Reply:
No, the Mono.error() seems to always return a json serialized body.  That is why bad.json fails.  So, we'll have to update webv to only look for status code.